Moved all advanced/dev documentation

vic10us opened the pull request:
Trying to clean up and simplify the README and related documentation so that users are less confused and developers/experts can still find their way.

jeffeb3 commented on src/docs/WSL_BUILD.md:
````diff
@@ -0,0 +1,25 @@
+- Install Python 
+```
+sudo apt update
+sudo apt install python3 python3-pip
````
You install pip, but not platformio. We need platformio to run build-for-machine.

jeffeb3 replied:
I would suggest installing this in a virtual env, but that may be more personal preference. I don't like my projects messing with the global python path and the packages in there. Python gets used all over my OS, so I don't want anything affected by some dep installed by platformio. I am running ubuntu natively on my laptop.

```
virtualenv -p python3 .venv
source .venv/bin/activate
pip install platformio
```
And then anytime you start a new session:
```
source .venv/bin/activate
```

vic10us replied:
Isn't platformio installed automatically from the build script? Let me check that...

vic10us replied:
It would appear that you are right... again. :) I'll add the necessary changes to the doc.

jeffeb3 commented on src/docs/WSL_BUILD.md:
````diff
@@ -0,0 +1,25 @@
+sudo apt update
+sudo apt install python3 python3-pip
+```
+- Add the following lines to your ~/.bashrc file: 
````
This can also lead to confusion. This is going to change everything you run to be python3 instead of python2. It is a very broad change.

vic10us replied:
sure, except that your build script is specifically looking for the aliased version of python, which doesn't exist unless you add the alias. I can add this detail to the doc so they can decide if it is what they need/want.

jeffeb3 replied:
I don't think the scripts even call python. It is only platformio that does. IIRC, platformio only works in python3. This is another reason I always use virtualenv. Once you set it, it will be python3 whenever you source that activation.

vic10us replied:
I'll test it without the aliases and if it works, I'll remove it from the docs alltogether.

jeffeb3 commented on src/docs/WSL_BUILD.md:
````diff
@@ -0,0 +1,25 @@
+```
+- Checkout Marlin
+```
+git clone git@github.com:MarlinFirmware/Marlin.git Marlin
````
```suggestion
git clone git@github.com:MarlinFirmware/Marlin.git Marlin -b bugfix-2.0.x --depth 1
```
This will just check out the most recent single commit, which is much faster. Then you won't need these next three lines.

vic10us replied:
Good catch! I'll make the change right away.

jeffeb3 replied:
When I work on this, I constantly rm -rf that directory and clone it again. It's nice to not have to download the whole history.

vic10us replied:
My approach is to do
```
git checkout .
git reset --hard
git clean -f
```

jeffeb3 commented on src/docs/README.md:
```diff
@@ -0,0 +1,44 @@
+
+## Local builds
+
+Don't want to use github workflows, or you want to test your builds before commiting and pushing to github, check out some recipes below. Missing something? Please feel free to contribute via PR.
```
```suggestion
If you want to use these scripts to configure Marlin on your local machine, then read the instructions here:
```

jeffeb3 reviewed (commented):
Looks good. Thank you for doing this.

vic10us commented:
I updated the configs to reflect the discussion points. I also renamed the WSL file to UBUNTU as it makes the assumption that the distro installed in WSL was Ubuntu and the instructions are identical. Although it's the default and the most likely distro to be installed it's a little presumptuous.

jeffeb3 commented on src/docs/UBUNTU_BUILD.md:
````diff
@@ -0,0 +1,42 @@
+```
+git clone git@github.com:MarlinFirmware/Marlin.git Marlin -b bugfix-2.0.x --depth 1
+cd Marlin
+git checkout bugfix-2.0.x
````
The -b already set the branch to bugfix-2.0.x.

vic10us replied:
oops... fixing

vic10us replied:
Fixed.

jeffeb3 commented:
LGTM. Thank you for doing this.
